### ISSUE TITLE
Strange issue with Orientation tag in XMP sidecar

### ISSUE THREAD

**Issue by narensankar0529** [RhetTbull/osxphotos#378]
This is a very strange issue that was seen when I imported pictures exported via osxphotos to Mylio.

Basically the orientation tag in the XMP is setup incorrectly. When exporting as HEIC, there is no Orientation tag in the HEIC Exif but the XMP records a Orientation of 6 which is rotate 90 CW.

When exporting as JPEG there is an Orientation flag in the Exif and is set to Normal while the XMP still records the Orientation as 6.

Mylio rotates the image when it is HEIC since the XMP overrides the Exif, whereas for JPEG it displays it correctly since Exif overrides XMP

Attaching a sample image that shows this issue. Not sure where osxphotos is getting the Orientation flag from to place in the XMP but it seems that does not match the actual image.

[Archive.zip](https://github.com/RhetTbull/osxphotos/files/5969723/Archive.zip)

**Comment by RhetTbull** [comment]
Were the photos rotated in Photos?  osxphotos reads the orientation from the Photos database but for some photos that are rotated in Photos, Photos does not update the orientation in the database, it stores the rotation in a separate .plist file with other adjustment data and osxphotos can't read this.  This is a known problem (though I'll admit it's not well documented...I'll open a new issue).  I cannot fix this at the moment as I can't read the adjustment data.  @neilpa has done some [work](https://github.com/neilpa/photohack) on using PhotoKit to get the orientation and other adjustment data and I plan to eventually add this capability to osxphotos but due to my extremely limited Objective-C skills, it'll take some time.

**Comment by narensankar0529** [comment]
No. The photos were not rotated manually. These are portrait mode pictures. I am not sure if all of them are portrait mode or not but a few that I checked are. I need to check more.

They are touched from capture.

Thanks!
—————

Naren Sankar

**Comment by narensankar0529** [comment]
They are untouched from capture.

Thanks!
—————

Naren Sankar

**Comment by RhetTbull** [comment]
Here's some additional debugging you can try that might provide useful info.  In Terminal, switch to the directory where you cloned the osxphotos source code for M1 testing then execute these commands:

```sh
$ git pull
$ python3 setup.py install
$ python3
```

Then in the python shell, execute the following commands, replacing "IMG_1234" with the name of one of the images in question.

```pycon
Python 3.8.1 (default, Jan  8 2020, 16:15:59)
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from osxphotos import PhotosDB
>>> photos = [p for p in PhotosDB().photos() if "IMG_1234" in p.original_filename]
>>> for p in photos:
...     print(p.original_filename, p.uuid, p.orientation, p.path)
...
IMG_1234.heic A8268455-E55F-4C22-BDC8-76B130B1F848 1 /Users/rhet/Pictures/Test-10.15.7.photoslibrary/originals/A/A8268455-E55F-4C22-BDC8-76B130B1F848.heic
>>> quit()
```

I'm curious what the orientation is for the image (3rd value printed above). The full path to the original image will the last value.  You can confirm it's the right photo by copying this value then doing this in the shell

```sh
$ open "/Users/rhet/Pictures/Test-10.15.7.photoslibrary/originals/A/A8268455-E55F-4C22-BDC8-76B130B1F848.heic"
```

**Comment by narensankar0529** [comment]
IMG_6269.HEIC 8FB82D80-6D28-42E8-8CF0-52FDAC5F1815 6 /Volumes/PhotoSync/nsankar/Photos Library.photoslibrary/originals/8/8FB82D80-6D28-42E8-8CF0-52FDAC5F1815.heic

**Comment by narensankar0529** [comment]
And if I export the HEIC from Photos directly it bakes in 6 in the EXIF. But osxphotos doesn't have the Orientation tag in the HEIC it exports, and in the JPEG it exports the Orientation tag is 1. Of course this is the same behavior of Photos as well. If I export JPEG it also sets the Orientation tag to 1. 

So not sure why there is this difference of behavior for HEIC vs JPEG. The only diff with Photos export and osxphotos export is that osxphotos doesn't place the Orientation tag in the EXIF.

**Comment by RhetTbull** [comment]
>And if I export the HEIC from Photos directly it bakes in 6 in the EXIF. But osxphotos doesn't have the Orientation tag in the HEIC it exports, 

I don't understand what could cause this behavior.  Are you using "Export unmodified original..." in Photos?  If so, that should match osxphotos as osxphotos simply copies the photo out of the library.

In my tests with a portrait mode HEIC, the original HEIC has orientation of 6 and the JPEG has no orientation set.   If I edit the photo in Photos, it changes the orientation to 1 which is what osxphotos will see.

I'm really not sure what the desired output is given that Photos treats the HEIC and JPEG differently.  Perhaps it is best to leave orientation out of the XMP and let whichever app is reading the image read it directly from exif?

**Comment by narensankar0529** [comment]
Yes. This would be my thinking as well. Photos seems to treat original and exported differently though in my test it output 6 when exporting modified original whereas osxphotos did not have any orientation. So it would imply Photos actually modified the original when exporting which is also weird.

So maybe the best is to not write orientation tag to xmp and let whatever is in the image dictate the tag. Though hopefully it won’t cause issues elsewhere. This seems to be a Photos bug IMHO but the workaround should be not to include in XMP. This is also what photos does. The XMP from photos don’t have the Orientation tag.

Thanks!
—————

Naren Sankar

**Comment by RhetTbull** [comment]
Fixed in [v0.40.16](https://github.com/RhetTbull/osxphotos/releases/tag/v0.40.16) -- I removed orientation from the XMP file.
